InstabugSample/ios/InstabugSampleTests/InstabugSampleTests.m

```diff
@@ -26,9 +28,191 @@ - (void)testShowingSurveyWithToken {
+  
+  [self.instabugBridge setInvocationEvents:invocationEventsArr];
+  IBGInvocationEvent invocationEvents = 0;
+  for (NSNumber *boxedValue in invocationEventsArr) {
+    invocationEvents |= [boxedValue intValue];
+  }
+  XCTAssertEqual(IBGBugReporting.invocationEvents, invocationEvents);
+}
+
+- (void)testgiven$invoke_whenQuery_thenShouldCallNativeApi {
+  id mock = OCMClassMock([IBGBugReporting class]);
+  
+  OCMStub([mock invoke]);
+  [self.instabugBridge invoke];
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Test ME PLX"];
+  
+  [[NSRunLoop mainRunLoop] performBlock:^{
+    OCMVerify([mock invoke]);
+    [expectation fulfill];
+  }];
+  
+  [self waitForExpectationsWithTimeout:EXPECTATION_TIMEOUT handler:nil];
+}
+
+- (void) testgivenOptions$setInvocationOptions_whenQuery_thenShouldCallNativeApiWithArgs {
+  NSArray *invocationOptionsArr = [NSArray arrayWithObjects:  @(IBGBugReportingInvocationOptionEmailFieldHidden), nil];
+  
+  [self.instabugBridge setInvocationOptions:invocationOptionsArr];
+  IBGBugReportingInvocationOption invocationOptions = 0;
+  for (NSNumber *boxedValue in invocationOptionsArr) {
+    invocationOptions |= [boxedValue intValue];
+  }
+  XCTAssertEqual(IBGBugReporting.invocationOptions, invocationOptions);
+}
+
+- (void) testgivenInvocationModeAndOptiond$invokeWithModeOptions_whenQuery_thenShouldCallNativeApiWithArgs {
+  NSArray *invocationOptionsArr = [NSArray arrayWithObjects:  @(IBGBugReportingInvocationOptionEmailFieldHidden), nil];
+  IBGBugReportingInvocationOption invocationOptions = 0;
+  for (NSNumber *boxedValue in invocationOptionsArr) {
+    invocationOptions |= [boxedValue intValue];
+  }
+  IBGInvocationMode invocationMode = IBGInvocationModeNewBug;
+  id mock = OCMClassMock([IBGBugReporting class]);
+  OCMStub([mock invokeWithMode:invocationMode options:invocationOptions]);
+  [self.instabugBridge invokeWithInvocationModeAndOptions:invocationMode options:invocationOptionsArr];
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Test ME PLX"];
+  
+  [[NSRunLoop mainRunLoop] performBlock:^{
+    OCMVerify([mock invokeWithMode:invocationMode options:invocationOptions]);
+    [expectation fulfill];
+  }];
+  
+  [self waitForExpectationsWithTimeout:EXPECTATION_TIMEOUT handler:nil];
+}
+
+- (void) testgivenPreInvocationHandler$setPreInvocationHandler_whenQuery_thenShouldCallNativeApi {
+  id partialMock = OCMPartialMock(self.instabugBridge);
+  RCTResponseSenderBlock callback = ^(NSArray *response) {};
+  [partialMock setPreInvocationHandler:callback];
+  XCTAssertNotNil(IBGBugReporting.willInvokeHandler);
+  OCMStub([partialMock sendEventWithName:@"IBGpreInvocationHandler" body:nil]);
+  IBGBugReporting.willInvokeHandler();
+  OCMVerify([partialMock sendEventWithName:@"IBGpreInvocationHandler" body:nil]);
+}
+
+
+- (void) testgivenPostInvocationHandler$setPostInvocationHandler_whenQuery_thenShouldCallNativeApi {
```

Comment:
I suggest breaking this into 3 separate methods, each with their own stubs and OCMVerify, to make it easier to know which one of them fails.